docs/experiment_tracking.md

```diff
@@ -0,0 +1,88 @@
+# Experiment tracking in llama recipes. 
+
+Llama recipes has an optional support for tracking experiments using experiment tracker modules like Aimstack or WandB (Currently not supported).
+
+The experiment tracker can be enabled by passing an additional argument to the llama recipes code which loads and
+tracks all the training and evaluation metrics in the tracker along with the system metrics being tracker by these trackers.
+
+The addition argument can be passed as - 
+
+```
+$ torchrun --nnodes 1 --nproc_per_node 1 examples/finetuning.py \
+           --model_name MODEL_NAME --output_dir OUTPUT_DIR \
+           --use_peft --peft_method lora --pure_bf16 \
+           --tracker aim
+```
+
+List of trackers integrated into llama recipes along with their support instructions are below.
+
+## Aimstack (https://aimstack.io/)
+
+Current version of the code supports Aimstack tracking via the trackers module.
+
+One needs to install Aimstack separately if need to use aimstack as a tracker.
+
+### Installation
+
+Aim can be installed by running
+
+```
+$ pip install aim==3.17.5
+```
+
+### Aim setup
+
+Aimstack is a self hosted open source component so a small setup is needed before running Aimstack.
+After installing Aim you need to
+
+1. Setup a local aim repo where the experiments will be tracked by running.
+
+```
+$ mkdir aimdb
+$ cd aimdb
+$ aim init
+```
+
+2. Start the Aimstack UI where you can visualize your runs. (run inside the aimdb folder)
+
+```
+$ aim up
+```
+
+You can visit the URL shown in the terminal (e.g. http://127.0.0.1:43800) to open Aimstack UI.
+
+### Running llama-recipes with Aim as a tracker.
+
+You can run the code with aimstack tracker by passing `--tracker aim` as an additional argument.
+
+When specifying Aim as a tracker you need to tell your code to point to the same aim db which we setup
+to track our experiments on our UI. You can do that by passing 
+
+`--aim_config.repo <path-to-aimdb>`
+
+You can categorize your expriments by passing a different name for each one in the command line using
+
+`--aim_config.experiment my_llama_experiment`
+
+Final command with all this would look like,
+
+```
+torchrun --nnodes 1 --nproc_per_node 1 examples/finetuning.py \
+           --model_name $MODEL_NAME --output_dir $OUTPUT_DIR \
+           --use_peft --peft_method lora --pure_bf16 \
+           --enable_fsdp --use_fast_kernels --num_epochs 1 --batch_size_training 1\
+           --tracker aim --aim_config.repo $AIMSTACK_DB \
+           --aim_config.experiment llama7b-finetuning
+```
+
+### Example screenshots.
+
+This is how the runs would look like, 
```

Comment:
nit: either "how the runs look" or "what the runs look like" not "how the runs look like"

When referring to a code snippet or image, it might also be better to be consistent in terms of punctuation. some end in a comma, some a period, some nothing. Maybe a colon would be better?